WebDriverAgentLib/Categories/XCUIDevice+FBHelpers.m

```diff
@@ -121,9 +125,28 @@ - (NSData *)fb_screenshotWithError:(NSError*__autoreleasing*)error
 #endif
 }
 
-- (NSData *)fb_rawScreenshotWithQuality:(NSUInteger)quality rect:(CGRect)rect error:(NSError*__autoreleasing*)error
+- (NSData *)fb_rawScreenshotWithQuality:(NSUInteger)quality rect:(CGRect)rect error:(NSError*__autoreleasing*) error
 {
-  return [XCUIScreen.mainScreen screenshotDataForQuality:quality rect:rect error:error];
```

Comment:
the `quality` here is the value in range 0..2. Also, non-zero value means the picture data should be encoded as jpeg.

Reply:
What do you mean by `Also, non-zero value means the picture data should be encoded as jpeg`. Does appium expect it to be jpeg or does the api return jpeg? I know that JsonWireProtocol returns base64 encoded PNG so I'm confused atm

Reply:
I mean png is lossless, so changing quality for such format literally has no effect. That is why we take the screenshot as JPEG and then covert it to PNG before sending to the client

Reply:
Now I get it. Makes sense. I will get a jpeg from here than

Reply:
this is not really a solution. The `_XCT_requestScreenshotOfScreenWithID` method only allows to explicitly set the picture type. Also, the quality value there is expected to be in range 0..100. I would rather keep the previous `screenshotDataForQuality` call to properly maintain the backward compatibility

Reply:
Done